src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -378,6 +379,8 @@ def generate_python_code(
             target_func.__annotations__ = annotations
     else:
         pickled_func = pickle_function(func)
+
+    pickled_by_reference = b"cloudpickle" not in pickled_func
```

Comment:
why do we use this to determine whether pickling by reference? What if cloudpickle is also imported in the inline python code of the SQL? 
```
create procedure ...
....
AS $$
import cloudpickle
...
$$
```

Reply:
This is a good point, let me see if I could find a better check.

Reply:
From what I tested, any reference to cloudpickle in the function would not show up as binary string. We can also check for `cloudpickle_submodules` instead.